sensors/context.go

```diff
@@ -51,25 +52,25 @@ type SensorContext struct {
 	hostname        string
 
 	// httpClients holds the reference to HTTP clients for HTTP triggers.
-	httpClients map[string]*http.Client
```

Comment:
Define a generic struct in the utilities like below?

```golang
type StringKeyedMap[T any] struct {
	items map[string]T
	lock     *sync.RWMutex
}
```

Reply:
Cool. I'll make the changes. Thanks.